### ISSUE TITLE
First arquitecture of the emulator

### ISSUE THREAD

**Issue by santanor** [KVault/NESulator#3]
## **This gon be gud** 

Right, so what we need to do here is to broadly define the files that we're going to use. For example, I reckon we need: 
- `emulator.c` that holds the main, basic emulation loop and a pointer to everything that is happening, like the `cpu`, `ppu`, and [`apu`](https://thepavlovictoday.com/wp-content/uploads/2018/04/apu.jpg) structs. I think this will also communicate with the UI once we have it in place, but we might have to split it. 
- `cpu.c` that defines the memory, registers, at some point we'll need to do the interrupts and everything. We have to decide what goes in here 
- `ppu.c` don't worry a lot about this one but we have to think, if the memory is shared between the `cpu` and the `ppu` (And probably the [`apu`](https://thepavlovictoday.com/wp-content/uploads/2018/04/apu.jpg) as well) Do we need a `memory` struct and a pointer to it? Or access it through the `cpu`? 

Basically lots of investigation but don't overextend it. Let's have the basic files and we'll move stuff around as we need it

**Comment by santanor** [comment]
It looks like we got most of this. Closing this issue for now.
